plugins/languages/treesitter/treesitter.nix

```diff
@@ -6,210 +6,169 @@
-          Register specific parsers to one or several filetypes.
-          The keys are the parser names and the values are either one or several filetypes.
-        '';
-        default = { };
-        example = {
-          cpp = "onelab";
-          python = [
-            "myFiletype"
-            "anotherFiletype"
-          ];
-        };
-      };
+    indent = {
+      enable = helpers.defaultNullOpts.mkBool true ''
+        Whether to enable tree-sitter based indentation.
+      '';
+    };
 
-      grammarPackages = mkOption {
-        type = with types; listOf package;
-        default = cfg.package.passthru.allGrammars;
-        description = "Grammar packages to install";
-      };
+    folding = {
+      enable = helpers.defaultNullOpts.mkBool true ''
+        Whether to enable tree-sitter based folding.
+      '';
+    };
 
-      moduleConfig = mkOption {
-        type = types.attrsOf types.anything;
-        default = { };
-        description = "This is the configuration for extra modules. It should not be used directly";
-      };
+    # TODO: default null nixgrammars
+    ensure_installed = helpers.defaultNullOpts.mkListOf types.str [ "all" ] ''
+      Either "all" or a list of languages to ensure installing.
+    '';
 
-      nixvimInjections = mkEnableOption "nixvim specific injections, like lua highlighting in extraConfigLua";
-    };
+    ignore_install = helpers.defaultNullOpts.mkListOf types.str [ ] ''
+      List of parsers to ignore installing (for "all")
+    '';
+
+    # TODO: default null nixgrammars
+    parser_install_dir = helpers.defaultNullOpts.mkStr "$XDG_DATA_HOME/nvim/treesitter" ''
+      Location of the parsers to be installed by the plugin (only needed when nixGrammars is disabled).
+      This default might not work on your own install, please make sure that $XDG_DATA_HOME is set if you want to use the default. Otherwise, change it to something that will work for you!
+    '';
   };
 
-  config =
-    let
-      tsOptions = {
-        highlight = {
-          inherit (cfg) enable;
-          disable = if (cfg.disabledLanguages != [ ]) then cfg.disabledLanguages else null;
+  extraOptions = {
+    # TODO: default null nixgrammars
+    gccPackage = helpers.mkPackageOption {
+      name = "gcc";
+      default = pkgs.gcc;
+      description = ''
+        Which package (if any) to be added as the GCC compiler.
+        This is required to build grammars if you are not using `nixGrammars`.
+        To disable the installation of GCC, set this option to `null`.
+      '';
+    };
+
+    grammarPackages = mkOption {
+      type = with types; listOf package;
+      default = cfg.package.passthru.allGrammars;
+      description = "Grammar packages to install";
+    };
 
-          custom_captures = if (cfg.customCaptures != { }) then cfg.customCaptures else null;
-        };
+    languageRegister = helpers.defaultNullOpts.mkAttrsOf (either str (listOf str)) { } ''
```

Comment:
I think using `types.coercedTo str toList (listOf str)` would be better.

This will allow defining strings or lists, but will merge multiple definitions as lists.

Although I don't think we do this elsewhere currently, it would allow multiple definitions to be merged when one def is a str and another is a list.

This is probably something we should consider in other places where we use `either`.